iceoryx2/src/service/mod.rs

```diff
@@ -695,11 +740,17 @@ pub trait Service: Debug + Sized + internal::ServiceInternal<Self> + Clone {
     /// Defines the thread-safety policy of the service. If it is defined as
     /// [`MutexProtected`](iceoryx2_cal::arc_sync_policy::mutex_protected::MutexProtected), the
     /// [`Service`]s ports are threadsafe and the payload can be moved into threads. If it is set
-    /// to to [`SingleThreaded`](iceoryx2_cal::arc_sync_policy::single_threaded::SingleThreaded),
+    /// to [`SingleThreaded`](iceoryx2_cal::arc_sync_policy::single_threaded::SingleThreaded),
     /// the [`Service`]s ports and payload cannot be shared ([`Sync`]) between threads or moved
     /// ([`Send`]) into other threads.
     type ArcThreadSafetyPolicy<T: Send + Debug>: ArcSyncPolicy<T>;
 
+    /// Defines the construct used to store the management data of the blackboard service.
+    type BlackboardMgmt<KeyType: Send + Sync + Debug + 'static>: DynamicStorage<KeyType>;
```

Comment:
Not in this PR.

Could we adjust the `DynamicStorage` type alias to be a generic alias like `trait DynamicStorage<T>: DynamicStorage<T>` and use it here like `type BlackboardMgmt<...>: Self::DynamicStorage<KeyType>` or even better - remove the type alias for `BlackboardMgmt` and `BlackboardPayload` (when we generalize the SharedMemory type alias as well).

It would have the big advantage that an inconsistent service variant would not be creatable. An inconsistent variant would be for instance a local variant that uses by accident posix shared memory for the blackboard.
The idea is to define the abstract concept implementation ones and then use it everywhere else.

Could you please track this in some issue.

Reply:
I created #882. I'm not sure if I fully understand your idea, so please have a look and adjust the issue if necessary.